### ISSUE TITLE
"[Vue Router warn]: No match found for location with path" on ANY NuxtLink with an external url

### ISSUE THREAD

**Issue by isaackearl** [nuxt/nuxt#19259]
### Describe the feature

right now in Nuxt 3.2.2, any usage of NuxtLink to an external url gives a warning "[Vue Router warn]: No match found for location with path <path>".  

Even if you look at the example provided in the documentation and you look at the developer console, you'll see the warnings once the page loads up
https://nuxt.com/docs/examples/routing/nuxt-link#nuxtlink

This is unintuitive, because obviously external links will not be available for the underlying vue-router and it won't know about them.  The project I am working on is a conversion of a legacy site, and so because of that we have to link to external pages ALOT within the Nuxt part of the application to jump back and forth with the legacy site.  That means every SPA page has tons of warnings in the console.  

The documentation shows examples of using NuxtLink for both internal and external links, and it even has an "external" property for edge cases.  It would be really great if we could somehow disable those warnings when using external links or manually setting the external property on the NuxtLink component.

Right now the alternative for me is to render two completely different components, one that is a NuxtLink, and one that is a simple anchor tag.  This works, but isn't as neat as just using NuxtLink as it is currently documented.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by bitbytebit1** [comment]
I'm seeing this behaviour on nuxt 3.12.1

**Comment by tewshi** [comment]
same here, in my case the `:to` props is a `mailto:` link, changed it to native `a` tag

**Comment by manchenkoff** [comment]
Yep, also with 3.12 noticed this issue. Here is an example that generates the warning:
```vue
<NuxtLink
        to="https://github.com/manchenkoff"
        external
        target="_blank"
        class="underline underline-offset-4"
      >
        @manchenkoff
</NuxtLink>
```

Regardless of `external` value, it keeps showing warning. Also tried to use `href` instead of `to`, didn't help 😞

**Comment by danielroe** [comment]
This should be resolved in v3.12.2.

**Comment by manchenkoff** [comment]
I've just upgraded and with 3.12.2 issue is gone, thanks @danielroe!

**Comment by rocketiscool** [comment]
> This should be resolved in v3.12.2.

It only fixed with target="_blank" not external but on the same page, for example when trying to href to a route that utilises the server API at /server/routes/auth/discord
